apps/src/code-studio/progressRedux.js

```diff
@@ -241,6 +242,37 @@ export const levelsByLesson = state => (
 );
 
 /**
+ * Groups lessons (aka stages) according to category.
+ * @returns {Object[]}
+ * {string} Object.name
+ * {string[]} Object.lessonNames
+ * {Object[]} Object.stageLevels
+ */
+export const categorizedLessons = state => {
+  let byCategory = {};
+
+  const allLevels = levelsByLesson(state);
+
+  state.stages.forEach((stage, index) => {
+    const category = stage.flex_category;
+    const lessonName = stage.name;
+    const stageLevels = allLevels[index];
+
+    byCategory[category] = byCategory[category] || {
+      category,
+      lessonNames: [],
+      levels: []
+    };
+
+    byCategory[category].lessonNames.push(lessonName);
+    byCategory[category].levels.push(stageLevels);
+  });
+
+  // We want to return an array of categories
+  return Object.keys(byCategory).map(key => byCategory[key]);
```

Comment:
Uh... in ES7-land this is just `Object.values(byCategory)`, right?  Maybe `_.values(byCategory)` for now?